src/clarity-angular/popover/signpost/if-open.directive.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {
+    Directive,
+    TemplateRef,
+    ViewContainerRef,
+    Input,
+    OnInit, Output, EventEmitter
+} from "@angular/core";
+
+import { IfOpenService } from "./if-open.service";
+
+@Directive({
+    selector: "[clrIfOpen]"
+})
+
+/**********
+ *
+ * @class IfOpenDirective
+ *
+ * @description
+ * A structural directive that controls whether or not the associated TemplateRef is instantiated or not.
+ * It makes use of a Component instance level service: IfOpenService to maintain state between itself and the component
+ * using it in the component template.
+ *
+ */
+export class IfOpenDirective implements OnInit {
+
+    /*********
+     * @property _open
+     *
+     * @description
+     * A property that holds the open or closed state for the template implementing this directive.
+     *
+     * @type {boolean}
+     * @private
+     */
+    private _open: boolean = false;
+
+    /*********
+     * @function open
+     *
+     * @description
+     * A setter for the directive property _open.
+     *
+     * @param value
+     */
+    @Input("clrIfOpen")
+    public set open( value: boolean ) {
+        value = !!value;
+        this._open = value;
+        this.openChange.emit(value);
```

Comment:
should we emit events on user inputs? I remember @youdz mentioned that we should avoid doing this but you should check with him.